feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-single-selection-base.component.ts

```diff
@@ -0,0 +1,157 @@
+import { ConfiguratorAttributeBaseComponent } from './configurator-attribute-base.component';
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Configurator } from '../../../../core/model/configurator.model';
+import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
+import { ConfiguratorAttributeQuantityComponentOptions } from '../../quantity/configurator-attribute-quantity.component';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorPriceComponentOptions } from '../../../price/configurator-price.component';
+
+@Component({
+  template: '',
+})
+export class ConfiguratorAttributeSingleSelectionBaseComponent extends ConfiguratorAttributeBaseComponent {
```

Comment:
Can we make it as `abstract`?  Even `ConfiguratorAttributeBaseComponent` is `abstract` as well. There is no such instance of it, right? And use `Directive()` as the decorator.

Reply:
Ideally we prefer to use services instead of abstract components.